core/src/core/transaction.rs

```diff
@@ -365,6 +365,22 @@ impl FixedLength for TxKernelEntry {
 		+ secp::constants::AGG_SIGNATURE_SIZE;
 }
 
+/// Enum of possible tx weight verification options -
+///
+/// * As "transaction" checks tx (as block) weight does not exceed max_block_weight.
+/// * As "block" same as above but allow for additional coinbase reward (1 output, 1 kernel).
+/// * As "pool" skips weight check (pool as one big tx, but without max weight).
+///
+#[derive(Clone, Copy)]
+pub enum WeightVerificationType {
+	/// Tx represents a tx (max block weight, accounting for additional coinbase reward).
+	AsTransaction,
+	/// Tx represents a block (max block weight).
+	AsBlock,
+	/// Tx represents the txpool (no max weight).
```

Comment:
Perhaps a nit but as the pool isn't really a core entity, I would name this more along the lines of `Skip` or `Ignore`. We may have other validation sources that want that as well.

Reply:
`NoLimit` to make this explicit. `Ignore` is maybe ambiguous here, are we skipping the check or are we skipping this tx somehow?